integrations/zoom/events.go

```diff
@@ -0,0 +1,170 @@
+// with a progressively increasing delay (5s, 10s, 20s, 40s, 80s) between each retry.
+// (Based on: https://developers.zoom.us/docs/api/webhooks/)
+func (h handler) handleEvent(w http.ResponseWriter, r *http.Request) {
+	defer r.Body.Close()
+
+	// Check the request's headers and parse its body.
+	zoomEvent := h.checkRequest(w, r)
+	if zoomEvent == nil {
+		return
+	}
+
+	eventType, ok := zoomEvent["event"].(string)
+	if !ok || eventType == "" {
+		h.logger.Warn("received Zoom event without event type", zap.Any("event", zoomEvent))
+		common.HTTPError(w, http.StatusBadRequest)
+		return
+	}
+
+	// Extract account ID from the Zoom event payload.
+	payload, ok := zoomEvent["payload"].(map[string]any)
+	if !ok {
+		h.logger.Warn("received Zoom event with invalid payload structure",
+			zap.String("event_type", eventType),
+			zap.Any("event", zoomEvent))
+		common.HTTPError(w, http.StatusBadRequest)
+		return
+	}
+
+	accountID, ok := payload["account_id"].(string)
+	if !ok {
+		h.logger.Warn("received Zoom event without account ID",
+			zap.String("event_type", eventType),
+			zap.Any("event", zoomEvent))
+		common.HTTPError(w, http.StatusBadRequest)
+		return
+	}
+
+	// Transform the Zoom event into an AutoKitteh event.
+	akEvent, err := common.TransformEvent(h.logger, zoomEvent, eventType)
+	if err != nil {
+		common.HTTPError(w, http.StatusInternalServerError)
+		return
+	}
+
+	// Retrieve all the relevant connections for this event.
+	ctx := r.Context()
+	cids, err := h.vars.FindConnectionIDs(ctx, desc.ID(), accountIDVar, accountID)
+	if err != nil {
+		h.logger.Error("failed to find connection IDs",
+			zap.String("account_id", accountID),
+			zap.Error(err))
+		common.HTTPError(w, http.StatusInternalServerError)
+		return
+	}
+
+	// Dispatch the event to all connections for potential asynchronous handling.
+	common.DispatchEvent(ctx, h.logger, h.dispatch, akEvent, cids)
+}
+
+// checkRequest validates the Zoom webhook HTTP request by verifying the content type,
+// required headers, and signature and returns the parsed JSON payload if valid.
+// Otherwise it returns nil with an HTTP error response.
+func (h handler) checkRequest(w http.ResponseWriter, r *http.Request) map[string]any {
+	l := h.logger.With(
+		zap.String("url_path", r.URL.Path),
+		zap.String("event_type", r.Header.Get("Zoom-Event")),
+	)
+
+	// Check the request's HTTP headers.
+	ct := r.Header.Get("Content-Type")
+	if !strings.HasPrefix(ct, "application/json") {
+		l.Warn("incoming event: unexpected content type", zap.String("content_type", ct))
+		common.HTTPError(w, http.StatusBadRequest)
+		return nil
+	}
+
+	signature := r.Header.Get("x-zm-signature")
+	timestamp := r.Header.Get("x-zm-request-timestamp")
+
+	if signature == "" || timestamp == "" {
+		l.Warn("incoming event: missing Zoom verification headers")
+		common.HTTPError(w, http.StatusBadRequest)
+		return nil
+	}
+
+	body, err := io.ReadAll(r.Body)
```

Comment:
```suggestion
	// Read the request's JSON body, up to 8 MiB, to prevent DDoS attacks.
	body, err := io.ReadAll(http.MaxBytesReader(nil, r.Body, 1<<23))
```